lib/rails-authorization-plugin/lib/publishare/identity.rb

```diff
@@ -15,88 +13,31 @@
 #
 module Authorization
   module Identity
+    VALID_PREPOSITIONS = %w[of for in on to at by].freeze
+    BOOLEAN_OPS = %w[not or and].freeze
+    VALID_PREPOSITIONS_PATTERN = VALID_PREPOSITIONS.join('|')
     module UserExtensions
       module InstanceMethods # rubocop:todo Style/Documentation
         def method_missing(method_sym, *args)
           method_name = method_sym.to_s
           authorizable_object = args.empty? ? nil : args[0]
 
           base_regex = 'is_(\\w+)'
-          fancy_regex = base_regex + "_(#{Authorization::Base::VALID_PREPOSITIONS_PATTERN})"
+          fancy_regex = base_regex + "_(#{Authorization::Identity::VALID_PREPOSITIONS_PATTERN})"
           is_either_regex = '^((' + fancy_regex + ')|(' + base_regex + '))'
-          base_not_regex = 'is_no[t]?_(\\w+)'
-          fancy_not_regex = base_not_regex + "_(#{Authorization::Base::VALID_PREPOSITIONS_PATTERN})"
-          is_not_either_regex = '^((' + fancy_not_regex + ')|(' + base_not_regex + '))'
 
-          if method_name =~ Regexp.new(is_either_regex + '_what$')
-            role_name = $3 || $6
-            has_role_for_objects(role_name, authorizable_object)
-          elsif method_name =~ Regexp.new(is_not_either_regex + '\?$')
-            role_name = $3 || $6
-            not is_role?(role_name, authorizable_object)
-          elsif method_name =~ Regexp.new(is_either_regex + '\?$')
+          # matches is_role? and is_role_of?
+          if method_name =~ Regexp.new(is_either_regex + '\?$')
             role_name = $3 || $6
```

Comment:
This is from previous code, but maybe can help in other changes to replace $3 and $6 with a variable name with
the description of the content, or maybe a coment explaining what it is referring in the regexp.

Reply:
This code is almost ready for complete removal, which is why I didn't touch it more (I was very tempted). I've already got the replacement methods working. But yes, agree with the suggestion in principal.